src/EnergyPlus/FluidProperties.cc

```diff
@@ -4744,7 +4744,12 @@ namespace FluidProperties {
                     break; // LoopC DO loop
                 }
             }
-            if (std::abs(RawConcData(HiIndex) - RawConcData(HiIndex - 1)) >= ConcToler) {
+
+            if (HiIndex == 1) {
+                for (LoopT = 1; LoopT <= NumOfTemps; ++LoopT) {
+                    InterpData(LoopT) = RawPropData(HiIndex, LoopT);
+                }
+            } else if (std::abs(RawConcData(HiIndex) - RawConcData(HiIndex - 1)) >= ConcToler) {
```

Comment:
Yes, your new line for `HiIndex == 1` seems like it would indeed work better and not have an array out of bounds issue.  Good.